cni/pkg/iptables/iptables.go

```diff
@@ -333,6 +333,13 @@ func (cfg *IptablesConfigurator) AppendInpodRules(podOverrides config.PodLevelOv
 		)
 	}
 
+	iptablesBuilder.AppendRule(ChainInpodPrerouting, "nat",
+		"-p", "tcp",
+		"-m", "dscp", "--dscp", fmt.Sprintf("%#x", config.DSCPMagicMark),
```

Comment:
This seems to be the best way to do this since this modifies the ip header and doesn't have the same issues a mark on the skb would have.